<launch>
  <arg name="mav_name" default="dji_m100"/>
  <arg name="suffix" default="default"/>
  <arg name="world_name" default="tian_gong_zhu_ta"/>
  <arg name="paused" default="false"/>
  <arg name="gui" default="true"/>
  <arg name="use_sim_time" default="false"/>
  <arg name="x_offset" default="0"/>
  <arg name="y_offset" default="0"/>
  <arg name="z_offset" default="0"/>
  <arg name="yaw_offset" default="3.14"/>
  
  <include file="$(find dji_gazebo)/launch/world.launch">
    <arg name="world_name" value="$(arg world_name)" />
    <arg name="paused" value="$(arg paused)"/>
    <arg name="gui" value="$(arg gui)"/>
    <arg name="use_sim_time" value="$(arg use_sim_time)" />
  </include>

  <group ns="$(arg mav_name)">
    <include file="$(find dji_gazebo)/launch/spawn_mav.launch">
      <arg name="mav_name" value="$(arg mav_name)" />
      <arg name="model" value="$(find dji_description)/urdf/$(arg mav_name)_$(arg suffix).xacro" />
      <arg name="is_kinematic" value="true"/>
    </include>
  </group>

  <node name="dji_osdk_bridge" pkg="dji_simulator" type="dji_osdk_bridge_node">
    <param name="x_offset" value="$(arg x_offset)"/>
    <param name="y_offset" value="$(arg y_offset)"/>
    <param name="z_offset" value="$(arg z_offset)"/>
    <param name="yaw_offset" value="$(arg yaw_offset)"/>
  </node>

  <node name="image_timestamp" pkg="dji_simulator" type="image_timestamp_node">
    <remap from="in" to="/camera/image_raw" />
    <remap from="out" to="/camera0/image_raw" />
  </node>

  <node name="image_view" pkg="image_view" type="image_view" args="image:=/camera0/image_raw">
    <param name="autosize" type="bool" value="false"/>
  </node>

</launch>
